includes/class-freemius.php

```diff
@@ -20980,8 +20980,8 @@ private function _sync_plugin_license(
                         $this->switch_to_blog( $this->_storage->network_install_blog_id );
                     }
 
-                    // Show API messages only if not background sync or if paying customer.
-                    if ( ! $background || $this->is_paying() ) {
+                    // Show API messages only if paying customer.
+                    if ( $this->is_paying() ) {

```

Comment:
@msladu when there's an API connectivity issue for a non-background request we **do** want to show the error notice, regardless of the `$this->is_paying()` result. So unless I'm missing something, this change should be reverted.

Reply:
@vovafeldman You are right. 

So far, we are showing admin notice only for non background process or for paying customer. 

Now, we want to do API test for non background process also. Do we want to check only for paying customer or also want to check for non paying customer?

Reply:
I'm not sure I get your question, so perhaps I'll try to clarify all the cases:

* If it's a non-background sync request (i.e., `$background = false`), if there's a connectivity issue => show error notice immediately whether `is_paying()` is `true` or `false` (no retries mechanism or something like that, the user needs to see an error since they clicked on something that supposed to do something with the API).
* If it's a background request (i.e., `$background = true`) and `is_paying()` returns `false` (i.e., the user is not on a paid plan so even if API requests fail, there's no reason to bother the user since it's not mission-critical for the functionality of the product).
* If it's a background request  (i.e., `$background = true`) and `is_paying()` returns `true` that's where we want to show the error notice (since lack of connectivity can cause issues). But, this is also the case where we want to add the retries mechanism so in case it's a temporary connectivity issue we won't bother the user about it.

Hope this is clearer.

Reply:
Thanks @vovafeldman!

It clarifies!